dbt/adapters/bigquery/relation_configs/materialized_view.py

```diff
@@ -0,0 +1,165 @@
+    kms_key_name: Optional[str] = None
+    friendly_name: Optional[str] = None
+    description: Optional[str] = None
+    labels: Optional[Dict[str, str]] = None
+
+    @classmethod
+    def from_dict(cls, config_dict) -> "BigQueryMaterializedViewConfig":
+        kwargs_dict = {
+            "materialized_view_name": cls._render_part(
+                ComponentName.Identifier, config_dict.get("materialized_view_name")
+            ),
+            "schema_name": cls._render_part(ComponentName.Schema, config_dict.get("schema_name")),
+            "database_name": cls._render_part(
+                ComponentName.Database, config_dict.get("database_name")
+            ),
+            "cluster_by": config_dict.get("cluster_by"),
+            "partition_by": config_dict.get("partition_by"),
+            "enable_refresh": config_dict.get("enabled_refresh"),
+            "refresh_interval_minutes": config_dict.get("refresh_interval_minutes"),
+            "hours_to_expiration": config_dict.get("hours_to_expiration"),
+            "max_staleness": config_dict.get("max_staleness"),
+            "allow_non_incremental_definition": config_dict.get(
+                "allow_non_incremental_definition"
+            ),
+            "kms_key_name": config_dict.get("kms_key_name"),
+            "friendly_name": config_dict.get("friendly_name"),
+            "description": config_dict.get("description"),
+            "labels": config_dict.get("labels"),
+        }
+
+        materialized_view: "BigQueryMaterializedViewConfig" = super().from_dict(kwargs_dict)  # type: ignore
+        return materialized_view
+
+    @classmethod
+    def parse_model_node(cls, model_node: ModelNode) -> dict:
+        config_dict = {
+            "materialized_view_name": model_node.identifier,
+            "schema_name": model_node.schema,
+            "database_name": model_node.database,
+            "cluster_by": model_node.config.extra.get("cluster_by"),
+            "partition_by": model_node.config.extra.get("partition_by"),
+            "partition_expiration_date": model_node.config.extra.get("partition_expiration_date"),
+            "refresh_interval_minutes": model_node.config.extra.get("refresh_interval_minutes"),
+            "hours_to_expiration": model_node.config.extra.get("hours_to_expiration"),
+            "max_staleness": model_node.config.extra.get("max_staleness"),
+            "allow_non_incremental_definition": model_node.config.extra.get(
+                "allow_non_incremental_definition"
+            ),
+            "kms_key_name": model_node.config.extra.get("kms_key_name"),
+            "friendly_name": model_node.config.extra.get("friendly_name"),
+            "description": model_node.config.extra.get("description"),
+            "labels": model_node.config.extra.get("labels"),
+        }
+
+        autorefresh_value = model_node.config.extra.get("enabled_refresh")
+        if autorefresh_value is not None:
```

Comment:
We should create this as a utility method, at least within `dbt-bigquery`. The logic for the string branch could also be simplified a bit, e.g.:

```python
def bool_setting(value: Optional[Any] = None) -> Optional[bool]:
    if value is None:
        return None
    elif isinstance(value, bool):
        return value
    elif isinstance(value, str):
        if value.lower() in ["true", "false"]:
            return bool(value)
        else:
            raise ValueError("...")
    else:
        raise TypeError("...")
```

Reply:
I added a file and implemented this method

Reply:
relevant: https://github.com/dbt-labs/dbt-bigquery/pull/914#discussion_r1336273529